syntax = "proto3";

package book;

service BookService {
  rpc GetBookById (BookIdRequest) returns (BookResponse);
  rpc GetAllBooks (Empty) returns (BooksResponse);
}

message BookIdRequest {
  int32 id = 1;
}

message Book {
  int32 id = 1;
  string title = 2;
  string author = 3;
  string published_date = 4;
}

message BookResponse {
  Book book = 1;
}

message BooksResponse {
  repeated Book books = 1;
}
